Distinguish components from external add-ons

shannonbradshaw opened the pull request:
This fixes #2536 I'm expecting to make small fixes depending on reviews.

shannonbradshaw commented:
cc: @RFMVasconcelos @cvenets

thesuperzapper commented:
@shannonbradshaw, looks good but I think we need to update `content/en/_redirects` to prevent dead links.

@RFMVasconcelos any other thoughts, or is this good to merge (after those redirects), so we can close #2536

RFMVasconcelos commented on content/en/docs/components/training/_index.md:
```diff
@@ -1,5 +1,5 @@
 +++
-title = "Training Operators"
+title = "Frameworks for Training"
```
We made this change recently in the opposite direction. I believe that `Training operators` is more accurate.

shannonbradshaw replied:
I'll fix it.

RFMVasconcelos reviewed (commented):
Thank you for this work @shannonbradshaw! 

A couple of missing things:
1. redirects in `_redirects` file so that external links are not broken

shannonbradshaw commented:
> Thank you for this work @shannonbradshaw!
> 
> A couple of missing things:
> 
> 1. Redirects in `_redirects` file so that external links are not broken
> 2. Do a search for internal links and update them accordingly not to break them upon these change
> 
> Do you think you could take on this extra work?

Yes. I'm checking through to do redirects.

shannonbradshaw commented:
@RFMVasconcelos @thesuperzapper, thanks for reviewing! I believe I've addressed your comments, including redirects and internal links. I did test and I believe the redirects and changes I made are working; however, given the tedious nature of these changes it's possible I missed something. Please let me know if you notice any further changes that need to be made.

thesuperzapper commented:
@shannonbradshaw can you resolve the merge conflict? 

@RFMVasconcelos I think this is a pretty good first pass, once the merge conflict is resolved, I think you can `/approve`

shannonbradshaw commented:
I broke this PR while attempting to rebase. Replacing this with a clean PR: #2630
